src/states/elder/mod.rs

```diff
@@ -332,23 +334,40 @@ impl Elder {
 
     // Connect to all neighbour elders we are not yet connected to and disconnect from peers that are no
     // longer members of our section or elders of neighbour sections.
-    fn update_neighbour_connections(&mut self, change: EldersChange, outbox: &mut dyn EventBox) {
+    fn update_neighbour_connections(&mut self, change: EldersChange, _outbox: &mut dyn EventBox) {
         if self.chain.prefix_change() == PrefixChange::None {
-            for pub_id in change.removed {
+            for p2p_node in change.removed {
                 // The peer might have been relocated from a neighbour to us - in that case do not
                 // disconnect from them.
-                if self.chain.is_peer_our_member(&pub_id) {
+                if self.chain.is_peer_our_member(p2p_node.public_id()) {
                     continue;
                 }
 
-                self.disconnect(&pub_id);
+                self.disconnect(p2p_node.public_id());
             }
         }
 
-        for pub_id in change.added {
-            let src = Authority::Node(*self.name());
-            let dst = Authority::Node(*pub_id.name());
-            let _ = self.send_connection_request(pub_id, src, dst, outbox);
+        for p2p_node in change.added {
+            let pub_id = *p2p_node.public_id();
+            if !self.peer_map.has(&pub_id) {
+                self.peer_map
+                    .insert(pub_id, p2p_node.connection_info().clone());
+                self.send_direct_message(&pub_id, DirectMessage::ConnectionResponse);
```

Comment:
Do we still need to do this? The neighbours are going to receive a `NeighbourInfo` from us which would also contain our `P2pNode`, so they will insert us into their `peer_map` when they process it.

Reply:
In the follow up PR we're going to try do remove the `ConnectionResponse` messages entirely, so leaving this as is for now